contract/src/msgs/data_requests/execute/commit_result.rs

```diff
@@ -18,15 +19,28 @@ impl ExecuteHandler for execute::commit_result::Execute {
 
         let resp = Response::new().add_attribute("action", "commit_data_result").add_event(
             Event::new("seda-commitment").add_attributes([
-                ("dr_id", self.dr_id),
+                ("dr_id", self.dr_id.clone()),
                 ("posted_dr_height", dr.height.to_string()),
                 ("commitment", self.commitment),
-                ("executor", self.public_key),
+                ("executor", self.public_key.clone()),
                 ("version", CONTRACT_VERSION.to_string()),
             ]),
         );
         state::commit(deps.storage, env.block.height, dr_id, dr)?;
-        Ok(resp)
+
+        let refund_msg = MsgRefundTxFee {
+            authority: env.contract.address.to_string(),
+            dr_id: self.dr_id,
+            public_key: self.public_key,
+        };
+        let mut vec = Vec::new();
+        refund_msg.encode(&mut vec).unwrap();
```

Comment:
We should handle this unwrap.

If you want my help let me know :)

Reply:
Might also be worth using the `Bytes` type, re-exported from the proto common crate. Would make the contract smaller & cheaper. Can be done in a follow up pr tho/I can help.

Reply:
Probably good to create a helper function to create the message?